packages/compartment-mapper/src/archive.js

```diff
@@ -204,27 +205,38 @@ const renameSources = (sources, compartmentRenames) => {
 };
 
 /**
- * @param {ArchiveWriter} archive
  * @param {Sources} sources
  */
-const addSourcesToArchive = async (archive, sources) => {
+export const locationsForSources = function* (sources) {
```

Comment:
are generators ok? I don't see a lot of them around the code base

Reply:
The one place I interacted with one it was annoying to debug, but no other complaints form me. Especially here, where they're used in a loop not passed to Array.from or Object.fromEntries